feat: add cloud secrets manager docs (#537)
kestra-io/docs: content/docs/09.administrator-guide/01.configuration/03.enterprise-edition/secrets/google-secret-manager/index.md

---
title: Google Secret Manager configuration
---

> This integration will store secrets in [Google Secret Manager](https://cloud.google.com/secret-manager). Kestra authentications must have [roles/secretmanager.admin](https://cloud.google.com/secret-manager/docs/access-control) permission.

Reviewer: ```suggestion
This integration will store secrets in [Google Secret Manager](https://cloud.google.com/secret-manager). Kestra authentication must have [roles/secretmanager.admin](https://cloud.google.com/secret-manager/docs/access-control) permission.
```

## Authentication

> You can handle authentication to Google Cloud in multiple ways:

Reviewer: ```suggestion
You can authenticate the Secrets Manager with Google Cloud in multiple ways:
```

> - By setting the serviceAccount property that must contain the service account JSON content.

Reviewer: ```suggestion
- By setting the `serviceAccount` property that must contain the contents of the service account JSON key file.
```

> - By setting the GOOGLE_APPLICATION_CREDENTIALS environment variable on the nodes running Kestra. It must point to an application credentials file. Warning: it must be the same on all worker nodes and can cause some security concerns.

Reviewer: ```suggestion
- By setting the `GOOGLE_APPLICATION_CREDENTIALS` environment variable on the nodes (or server) running Kestra. It must point to a JSON credentials file. Keep in mind that you'd need to use the same variable on all worker nodes and that this authentication method can cause some security concerns.
```
